### PR TITLE
Use Nightly Clippy + bug fixes

### PR DESCRIPTION
**Pull request by jplatte**

### CODE REVIEW

File: crates/matrix-sdk-base/src/client.rs [hunk 1]
```diff
@@ -87,7 +87,9 @@ pub struct BaseClient {
     olm: Arc<Mutex<Option<OlmMachine>>>,
     #[cfg(feature = "encryption")]
     cryptostore: Arc<Mutex<Option<Box<dyn CryptoStore>>>>,
+    #[cfg(feature = "encryption")]
```
Comment (poljar) [inline crates/matrix-sdk-base/src/client.rs:90]: Is this right? The `store_path` is used for the sled state store as well, which can be enabled without having `encryption` enabled.

File: crates/matrix-sdk/examples/get_profiles.rs [hunk 2]
```diff
@@ -7,6 +7,7 @@ use matrix_sdk::{
 use url::Url;
 
 #[derive(Debug)]
+#[allow(dead_code)]
```
Comment (poljar) [inline crates/matrix-sdk/examples/get_profiles.rs:10]: Why is this dead if it's clearly used?

**Review by poljar — CHANGES_REQUESTED** [review]
Mostly looks good, two things seem to be a bit fishy?

File: crates/matrix-sdk/examples/get_profiles.rs (hunk 2 above)
Comment (jplatte) [reply crates/matrix-sdk/examples/get_profiles.rs:10]: It's written, but never read except for through `Debug`, which Nightly rustc no longer considers as a "real" use.

File: crates/matrix-sdk-base/src/client.rs (hunk 1 above)
Comment (jplatte) [reply crates/matrix-sdk-base/src/client.rs:90]: I thought I checked all usages. I'll have another look.

Comment (jplatte) [reply crates/matrix-sdk-base/src/client.rs:90]: I checked: With the `sled_state_store` feature, only `config.store_path` is used (which I did not touch), not `client.store_path`.

Comment (poljar) [reply crates/matrix-sdk-base/src/client.rs:90]: Oh sorry, I seem to have confused those two.

**Review by poljar — APPROVED** [review]

**Pull request merged**